<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>636</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>TOTAL By LukeSkyD</string>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QFrame" name="frameStatus">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_7">
       <item>
        <layout class="QHBoxLayout" name="layoutKit">
         <item>
          <widget class="QPushButton" name="bHOME">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>HOME</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="lKit">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>ADB &amp; Fastboot:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="lPath">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Not found. Manually select the ADB and Fastboot directory</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="bKit">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Change folder</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="layoutDevice">
         <item>
          <widget class="QPushButton" name="bRefresh">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>REFRESH</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="lInd">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Device:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="lDevice">
           <property name="text">
            <string>Not found</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QFrame" name="frameActions">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>400</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item alignment="Qt::AlignHCenter|Qt::AlignTop">
        <widget class="QStackedWidget" name="stackedWidget">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>300</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>361</width>
           <height>471</height>
          </size>
         </property>
         <property name="currentIndex">
          <number>3</number>
         </property>
         <widget class="QWidget" name="p00">
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QLabel" name="lWelcome">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Welcome</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="bADB">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>320</width>
               <height>40</height>
              </size>
             </property>
             <property name="text">
              <string>Utilities</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="bBL">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>320</width>
               <height>40</height>
              </size>
             </property>
             <property name="text">
              <string>Fastboot Flasher</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="bEDL">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>320</width>
               <height>40</height>
              </size>
             </property>
             <property name="text">
              <string>Unbrick</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="pBL_00">
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QLabel" name="lBL">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Fastboot Flasher</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="bUNOS">
             <property name="minimumSize">
              <size>
               <width>320</width>
               <height>40</height>
              </size>
             </property>
             <property name="text">
              <string>Update NOS
(no data wipe, flashes one slot)</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pRNOS">
             <property name="minimumSize">
              <size>
               <width>320</width>
               <height>40</height>
              </size>
             </property>
             <property name="text">
              <string>Restore NOS
(data wipe, flash to both slots)</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pCROM">
             <property name="minimumSize">
              <size>
               <width>320</width>
               <height>40</height>
              </size>
             </property>
             <property name="text">
              <string>Flash Custom ROM</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pMFL">
             <property name="minimumSize">
              <size>
               <width>320</width>
               <height>40</height>
              </size>
             </property>
             <property name="text">
              <string>Manual Flash</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="pBL_01">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QLabel" name="lInstruct">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>140</height>
              </size>
             </property>
             <property name="text">
              <string>Firmware Flash

CHOOSE ONE FROM THE LIST OF SUPPORTED FILES:
- FULL OTA .zip FILE
- payload.bin IF ALREADY EXTRACTED
- FOLDER CONTAINING ALL THE .img FILES
(select one of the .img)
- .7z AND .7z.00x CONTAINING THE FW
</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QFrame" name="frameOptions">
             <property name="enabled">
              <bool>true</bool>
             </property>
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_6">
              <item>
               <layout class="QGridLayout" name="gridBLOptions">
                <item row="0" column="0">
                 <widget class="QCheckBox" name="cbWipe">
                  <property name="text">
                   <string>Data wipe</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="0">
                 <widget class="QCheckBox" name="cbAVB">
                  <property name="text">
                   <string>Disable Android verfied
boot</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QCheckBox" name="cbLogi">
                  <property name="text">
                   <string>Flash logical partitions</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QCheckBox" name="cb2Slo">
                  <property name="text">
                   <string>Flash both slots</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QCheckBox" name="cbDCheck">
                  <property name="text">
                   <string>Disable files integrity
check</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="lAdd">
             <property name="text">
              <string>Add files or folder here
(If multiple valid FWs are added, only the first one will be flashed)</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QPushButton" name="bAdd">
               <property name="text">
                <string>Add</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="bRm">
               <property name="text">
                <string>Remove</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QListWidget" name="listWidget"/>
           </item>
           <item>
            <widget class="QPushButton" name="bBLConf">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>40</height>
              </size>
             </property>
             <property name="text">
              <string>Confirm and Start flashing</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="pADB_00">
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_8">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout">
               <item>
                <widget class="QRadioButton" name="rbReboot">
                 <property name="text">
                  <string> Reboot to</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="cbReboot">
                 <item>
                  <property name="text">
                   <string>system</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>bootloader</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>recovery</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>fastbootd</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>edl</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <item>
                <widget class="QRadioButton" name="rbLUBL">
                 <property name="text">
                  <string>Lock/unlock the bootloader</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="cbLUBL">
                 <item>
                  <property name="text">
                   <string>lock</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>unlock</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_4">
               <item>
                <widget class="QRadioButton" name="rbLUCP">
                 <property name="text">
                  <string>Lock/unlock critical partitions</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="cbLUCP">
                 <item>
                  <property name="text">
                   <string>lock</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>unlock</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QPushButton" name="btnOptions">
               <property name="text">
                <string>Proceed</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="layExt">
             <item>
              <widget class="QLabel" name="lExt">
               <property name="text">
                <string>Extract partitions from payload.bin</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnExt">
               <property name="text">
                <string>Manage</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="layMBB">
             <item>
              <widget class="QLabel" name="lMBB">
               <property name="text">
                <string>Manage Magisk boot backup</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pushButton_4">
               <property name="text">
                <string>Manage</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuQT_infos">
    <property name="title">
     <string>About</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionAbout_QT"/>
   </widget>
   <addaction name="menuQT_infos"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionAbout_QT">
   <property name="text">
    <string>About QT</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
